### PR TITLE
Add plugin support on Search functionality (task #1900)

### PR DESCRIPTION
**Pull request by georgeconstantinou**

### CODE REVIEW

File: src/Controller/Component/SearchableComponent.php [hunk 1]
```diff
@@ -441,7 +443,7 @@ protected function _basicWhereStatement(array $data, $model)
             $fields = $this->getSearchableFieldProperties($model, $fields);
             foreach ($fields as $field => $properties) {
                 if (in_array($properties['type'], $this->_basicSearchFieldTypes)) {
-                    $result['OR'][$model . '.' . $field . ' LIKE'] = '%' . $data['query'] . '%';
+                    $result['OR'][$field . ' LIKE'] = '%' . $data['query'] . '%';
```
Comment (mamchenkov) [inline src/Controller/Component/SearchableComponent.php]: @georgeconstantinou what's the reason for this change?  Referencing the model seems like the right thing to do, to avoid ambiguity in the field names.

Comment (georgeconstantinou) [inline src/Controller/Component/SearchableComponent.php]: @mamchenkov the model now includes the plugin name `MyPlugin.MyModel` and breaks the query.

Comment (mamchenkov) [inline src/Controller/Component/SearchableComponent.php]: @georgeconstantinou OK, I'll merge for now.  But later it's better to clean up the model name and use it as a field prefix.  It will save us a lot of headache in the future.  The kind where SQL errors.
